sdk/anomalydetector/azure-ai-anomalydetector/samples/sample_multivariate_detect.py

```diff
@@ -19,200 +19,214 @@
 
-    def __init__(self, subscription_key, anomaly_detector_endpoint, data_source=None):
+    def __init__(self, subscription_key, anomaly_detector_endpoint):
         self.sub_key = subscription_key
         self.end_point = anomaly_detector_endpoint
 
         # Create an Anomaly Detector client
 
         # <client>
-        self.ad_client = AnomalyDetectorClient(AzureKeyCredential(self.sub_key), self.end_point)
+        self.ad_client = AnomalyDetectorClient(self.end_point, AzureKeyCredential(self.sub_key))
         # </client>
 
-        self.data_source = data_source
+    def list_models(self):
 
-    def train(self, start_time, end_time):
+        # List models
+        skip = 0
 
-        # Number of models available now
-        model_list = list(self.ad_client.list_multivariate_model(skip=0, top=10000))
-        print("{:d} available models before training.".format(len(model_list)))
+        next_link = None
+        model_list = []
 
-        # Use sample data to train the model
-        print("Training new model...(it may take a few minutes)")
-        data_feed = ModelInfo(start_time=start_time, end_time=end_time, source=self.data_source)
-        response_header = \
-            self.ad_client.train_multivariate_model(data_feed, cls=lambda *args: [args[i] for i in range(len(args))])[
-                -1]
-        trained_model_id = response_header['Location'].split("/")[-1]
+        while next_link != '':
+            r = self.ad_client.list_multivariate_model(skip=skip, top=10000)
+            next_link = r['nextLink'] 
+            model_list.extend(r['models'])
+            skip = skip + 10
+        return model_list
 
-        # Wait until the model is ready. It usually takes several minutes
-        model_status = None
+    def train(self, body):
 
-        while model_status != ModelStatus.READY and model_status != ModelStatus.FAILED:
-            model_info = self.ad_client.get_multivariate_model(trained_model_id).model_info
-            model_status = model_info.status
-            time.sleep(1)
+        # Number of models available now
+        try:
+            model_list = self.list_models()
+            print("{:d} available models before training.".format(len(model_list)))
+
+            # Use sample data to train the model
+            print("Training new model...(it may take a few minutes)")
+            response = self.ad_client.create_multivariate_model(body)
+            trained_model_id = response['modelId']
+            print("Training model id is ", trained_model_id)
+
+            ## Wait until the model is ready. It usually takes several minutes
+            model_status = None
+            model_info = None
+
+            while model_status != 'READY' and model_status != 'FAILED':
+                model_info = self.ad_client.get_multivariate_model(trained_model_id)
+                model_status = model_info['modelInfo']['status']
+                print("Model is", model_status)
+                time.sleep(1)
```

Comment:
time.sleep(30)? i think it is not necessary to check model status in such a high frequency